File: 백준/Silver/2961. 도영이가 만든 맛있는 음식/도영이가 만든 맛있는 음식.js
function solution(input) {
  const testCount = parseInt(input.shift(), 10);
  const taste = input.map((string) => string.split(" ").map(Number));

  let minDiff = Infinity;
  function getDiff(index, sour, bitter, count) {
    if (index === testCount) {
      if (count > 0) {
        minDiff = Math.min(minDiff, Math.abs(sour - bitter));
      }
      return;
    }

    getDiff(
      index + 1,
      sour * taste[index][0],
      bitter + taste[index][1],
      count + 1
    );

    getDiff(index + 1, sour, bitter, count);
  }

  getDiff(0, 1, 0, 0);

  return minDiff;
}

const input = [];
require("readline")
  .createInterface({
    input: process.stdin,
    output: process.stdout,
  })
  .on("line", (line) => {
    input.push(line);
  })
  .on("close", () => {
    console.log(solution(input));
    process.exit();
  });

File: 백준/Silver/2961. 도영이가 만든 맛있는 음식/도영이가 만든 맛있는 음식.py
import sys
input = sys.stdin.readline

def solution() :
  test_count = int(input())
  taste = [list(map(int, input().split())) for _ in range(test_count)]
  
  min_diff = float('inf')
  
  def get_diff (index, sour, bitter, count):
    nonlocal min_diff
    if index == test_count:
      if count > 0:
        min_diff = min(min_diff, abs(sour - bitter))
      return
      
    get_diff(index + 1, sour * taste[index][0], bitter + taste[index][1], count + 1)
    
    get_diff(index + 1, sour, bitter, count)
      
  get_diff(0,1,0,0)
    
  print(min_diff)
  
solution()